app/models/translation.rb

```diff
@@ -24,8 +24,17 @@ class Translation < ActiveRecord::Base
   # Maps from status symbol to integer representation for database:
   @@STATUS_TO_INT_HASH = @@STATUS_HASH.invert
 
-  def points
-    @@POINTS_HASH[self.status_symbol]
+  def points(since)
+    if since == 'month'
+      after = 1.month.ago
+    elsif since == 'year'
+      after = 1.year.ago
+    else
+      after = Time.new(0)
+    end
```

Comment:
Logically, it'll be better to take in since as a Time object, which means moving this switch logic into the controller. Should probably rename `since` to `from`.